tools/codegen/dest/lazy_ir.py

```diff
@@ -58,6 +73,37 @@ def aten_symbol(schema: LazyIrSchema) -> str:
         return f'c10::Symbol::fromQualString("aten::{schema.aten_name}")'
     return f'at::aten::{schema.aten_name}'
 
+def cached_shape_inference(func: NativeFunction, schema: LazyIrSchema,
+                           returns_length: int, all_types: List[NamedCType]) -> str:
+    # call the meta kernel if it exists, to compute output shape/dtype for our IR
+    if func.structured or func.structured_delegate is not None:
+        meta_out = """cached_shapes = shape_cache->Add(dag_hash, std::make_shared<std::vector<Shape>>(std::initializer_list<Shape>{Shape(out_meta.scalar_type(), out_meta.sizes().vec())}));"""
+        if returns_length > 1:
+            def this_shape(i: int) -> str:
+                return f"Shape(std::get<{i}>(out_meta).scalar_type(), std::get<{i}>(out_meta).sizes().vec())"
```

Comment:
why are we using full shapes here compared to only using a rank above?

Reply:
i'm not sure i understand what you mean.